modules/steam.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    mkDefault
+    mkIf
+    mkMerge
+    mkOption
+    types
+  ;
+  # ¯\_(ツ)_/¯
+  token = "steampal_stable_9a24a2bf68596b860cb6710d9ea307a76c29a04d";
+
+  # This script is used to hijack the call to `dbus-send` that the steam client
+  # will to to return to the desktop.
+  dbus-hijacker = (pkgs.writeShellScriptBin "dbus-send" ''
+    if echo "$@" | grep 'DisplayManager.*SwitchToUser.*doorstop.*plasma'; then
+      systemctl --user stop steam
+      exit 0
+    fi
+
+    # Snitches on unhandled calls to `dbus-send`
+    # echo dbus-send "$@" >> ~/dbus-hijacker.log
+
+    exec ${pkgs.dbus}/bin/dbus-send "$@"
+  '').overrideAttrs({ meta, ... }: { meta = meta // { piority = 9999; }; });
+
+  steam = pkgs.steam.override {
+    extraPkgs = pkgs: [ dbus-hijacker ];
+  };
+in
+{
+  options = {
+    jovian = {
+      steam = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Enables stopped-by-default systemd service to launch steam
+            in "PAL" user interface (steam deck interface).
+
+            Use `systemctl --user start steam` to launch.
+
+            Use `systemctl --user stop steam` to stop.
+          '';
+        };
+      };
+    };
+  };
+  config = mkMerge [
+    (mkIf config.jovian.steam.enable {
+      hardware.opengl.driSupport32Bit = true;
+      hardware.pulseaudio.support32Bit = true;
+      jovian.enableControllerUdevRules = true;
+
+      systemd.user.services."steam" = {
+        enable = true;
+        serviceConfig = {
+          KillMode = "process";
+          Restart = "always";
+          RestartSec = "1";
+          ExecStart = pkgs.writeShellScript "steam-pal-ui" ''
+            export PATH="${lib.makeBinPath ([
+              pkgs.gamescope
+              steam
+            ])}:$PATH"
+            # This forces the user in the steampal beta.
+            # Not ideal, as it's not reversed by disabling this service.
+            echo "${token}" > ~/.local/share/Steam/package/beta
+            exec gamescope \
+              --fullscreen \
```

Comment:
Add options for gamescope+steam

 - Make a submodule with all relevant gamescope options
 - Apply submodule as CLI options

Relevant options:

```
  --help                         show help message
  -w, --nested-width             game width
  -h, --nested-height            game height
  -r, --nested-refresh           game refresh rate (frames per second)
  -m, --max-scale                maximum scale factor
  -i, --integer-scale            force scale factor to integer
  -W, --output-width             output width
  -H, --output-height            output height
  -n, --nearest-neighbor-filter  use nearest neighbor filtering
  -U  --fsr-upscaling            use AMD FidelityFX™ Super Resolution 1.0 for upscaling
  --fsr-sharpness                FSR sharpness from 0 (max) to 20 (min)
  --cursor                       path to default cursor image
  -C, --hide-cursor-delay        hide cursor image after delay
  -e, --steam                    enable Steam integration
 --xwayland-count                create N xwayland servers

Nested mode options:
  -o, --nested-unfocused-refresh game refresh rate when unfocused
  -b, --borderless               make the window borderless
  -f, --fullscreen               make the window fullscreen

Embedded mode options:
  -O, --prefer-output            list of connectors in order of preference
```